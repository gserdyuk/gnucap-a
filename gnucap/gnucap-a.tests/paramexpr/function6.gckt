* logic functions

.param f1(x)  = '(x>0)*1'
.param f2(x)  = '(x<=0)*2'
.param f3(x,y)= 'x==y ? 0 : 1 '
.param f4(x,y)= 'x||y'
.param f5(x,y)= 'x&&y'

* expected 0
v10 10 0 dc 'f1(-1)'
* expected 0
v11 11 0 dc 'f1(0)'
* expected 1
v12 12 0 dc 'f1(0.1)'

* expected 2
v20 20 0 dc 'f2(-1)'
* expected 2
v21 21 0 dc 'f2(0)'
* expected 0
v22 22 0 dc 'f2(0.1)'

* expected 1
v30 30 0 dc 'f3(1,1.2)'
* expected 0
v31 31 0 dc 'f3(1.1,1.1)'

* expected 1
v40 40 0 dc 'f4(1,0)'
* expected 0
v41 41 0 dc 'f4(0,0)'

* expected 1
v50 50 0 dc 'f5(1,1)'
* expected 0
v51 51 0 dc 'f5(1,0)'



.print op v(v*) 

.op

.end
