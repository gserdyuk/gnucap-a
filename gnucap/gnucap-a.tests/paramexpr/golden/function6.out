Gnucap gnucap-arails  2009.12.07 RCS 26.136
The Gnu Circuit Analysis Package
Never trust any version less than 1.0
Copyright 1982-2009, Albert Davis
Gnucap comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome
to redistribute it under the terms of 
the GNU General Public License, version 3 or later.
See the file "COPYING" for details.
* logic functions 
#           v(v10)     v(v11)     v(v12)     v(v20)     v(v21)     v(v22)     v(v30)     v(v31)     v(v40)     v(v41)     v(v50)     v(v51)    
 27.        0.         0.         1.         2.         2.         0.         0.         1.         1.         0.         1.         0.        
